### PR TITLE
feat: initial worker framework

### PR DESCRIPTION
**Pull request by agaffney**

### CODE REVIEW

File: internal/config/config.go [hunk 1]
```diff
@@ -103,6 +109,11 @@ var globalConfig = &Config{
 		// TODO: pick a better location
 		Directory: "./.bluefin",
 	},
+	// The default worker config is somewhat conservative: worker count is set
+	// to half of the available logical CPUs
+	Worker: WorkerConfig{
+		Count: runtime.NumCPU() / 2,
```
Comment (wolf31o2) [inline internal/config/config.go:115]: This is cgroup aware?

File: internal/worker/worker.go [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+			defer m.workerWaitGroup.Done()
+			for {
+				// Check for worker shutdown
+				select {
+				case <-m.doneChan:
+					return
+				default:
+					break
+				}
+				// TODO
```
Comment (wolf31o2) [inline internal/worker/worker.go:56]: Is this a "mining" stub?

**Review by wolf31o2 — APPROVED** [review]
I have a couple questions, but this looks good to me 👍

File: internal/config/config.go (hunk 1 above)
Comment (agaffney) [reply internal/config/config.go:115]: No, it's apparently not...

File: internal/worker/worker.go (hunk 2 above)
Comment (agaffney) [reply internal/worker/worker.go:56]: Yes

**Pull request merged**